api/v1alpha3/awscluster_webhook.go

```diff
@@ -80,10 +85,38 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	allErrs = append(allErrs, r.validateCIDRBlocks()...)
+
 	return aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
 }
 
+func (r *AWSCluster) validateCIDRBlocks() []*field.Error {
+	var errs field.ErrorList
+
+	if r.Spec.Bastion.DisableIngressRules && len(r.Spec.Bastion.CIDRBlocks) > 0 {
+		errs = append(errs,
+			field.Forbidden(field.NewPath("spec", "Bastion", "CIDRBlocks"), "cannot be set if spec.Bastion.DisableIngressRules is true"),
+		)
+		return errs
+	}
+
+	re, _ := regexp.Compile(`^([0-9]{1,3}\.){3}[0-9]{1,3}(\/([0-9]|[1-2][0-9]|3[0-2]))?$`)
```

Comment:
Rather than using a regex here it might be better to use [net.ParseCIDR](https://golang.org/pkg/net/#ParseCIDR) instead, which would also make it easier to support ipv6 in the future as well.

Reply:
thanks @detiber ! I wasn't aware of `net.ParseCIDR` and will update the validation to take advantage of it.

Reply:
@detiber switched to `net.ParseCIDR`